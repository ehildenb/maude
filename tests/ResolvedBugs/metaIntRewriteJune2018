#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/metaIntRewriteJune2018.maude -no-banner -no-advise \
  > metaIntRewriteJune2018.out 2>&1

diff $srcdir/metaIntRewriteJune2018.expected metaIntRewriteJune2018.out > /dev/null 2>&1
