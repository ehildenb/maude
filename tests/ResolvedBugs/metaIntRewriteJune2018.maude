set show timing off .
set show advisories off .

***
***	Check that the meta-interpreter resets the next rule to try at each rewriteTerm() message.
***

load metaInterpreter

mod FOO is
  sort Foo .
  ops a b c : -> Foo .
  rl a => b .
  rl a => c .
endm

mod TEST is
  pr META-INTERPRETER .

  op me : -> Oid .
  op User : -> Cid .

  vars X Y Z : Oid .
  var T : Term .
  var S : Sort .
  var A : AttributeSet .
  var N : Nat .

  op prev:_ : Term -> Attribute [ctor] .

  rl < X : User | A > createdInterpreter(X, Y, Z) => < X : User | A >
       insertModule(Z, X, upModule('FOO, true)) .

  rl < X : User | A > insertedModule(X, Y) => < X : User | A >
       rewriteTerm(Y, X, unbounded, 'FOO, 'a.Foo) .

  rl  < X : User | none > rewroteTerm(X, Y, N, T, S) => < X : User | prev: T >
       rewriteTerm(Y, X, unbounded, 'FOO, 'a.Foo) .
endm

*** second rewriteTerm returns different result from first.
erew  <> 
< me : User | none > 
createInterpreter(interpreterManager, me, none) .
