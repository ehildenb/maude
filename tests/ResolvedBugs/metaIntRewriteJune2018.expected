==========================================
erewrite in TEST : <> < me : User | none > createInterpreter(
    interpreterManager, me, none) .
rewrites: 6
result Configuration: <> < me : User | prev: 'b.Foo > rewroteTerm(me,
    interpreter(0), 1, 'b.Foo, 'Foo)
Bye.
